master/buildbot/reporters/gerrit_verify_status.py

```diff
@@ -0,0 +1,198 @@
+
+        if baseURL.endswith('/'):
+            baseURL = baseURL[:-1]
+
+        self._http = yield httpclientservice.HTTPClientService.getService(
+            self.master, baseURL, auth=auth)
+
+        self.verification_name = verification_name or Interpolate('%(prop:buildername)s')
+        self.reporter = reporter or "buildbot"
+        self.abstain = abstain
+        self.category = category
+        self.startDescription = startDescription or 'Build started.'
+        self.endDescription = endDescription or 'Build done.'
+        self.verbose = verbose
+
+    def createStatus(self,
+                     change_id, revision_id, name, value, abstain=None,
+                     rerun=None, comment=None, url=None, reporter=None,
+                     category=None, duration=None):
+        """
+        Abstract the POST REST api documented here:
+        https://gerrit.googlesource.com/plugins/verify-status/+/master/src/main/resources/Documentation/rest-api-changes.md
+
+        :param change_id: The change_id for the change tested (can be in the long form e.g:
+            myProject~master~I8473b95934b5732ac55d26311a706c9c2bde9940 or in the short integer form).
+        :param revision_id: the revision_id tested can be the patchset number or
+            the commit id (short or long).
+        :param name: The name of the job.
+        :param value: The pass/fail result for this job: -1: fail 0: unstable, 1: succeed
+        :param abstain: Whether the value counts as a vote (defaults to false)
+        :param rerun: Whether this result is from a re-test on the same patchset
+        :param comment: A short comment about this job
+        :param url: The url link to more info about this job
+        :reporter: The user that verified this job
+        :category: A category for this job
+        "duration": The time it took to run this job
+
+        :return: A deferred with the result from Gerrit.
+        """
+        payload = {
+            'name': name,
+            'value': value}
+
+        if abstain is not None:
+            payload['abstain'] = abstain
+
+        if rerun is not None:
+            payload['rerun'] = rerun
+
+        if comment is not None:
+            payload['comment'] = comment
+
+        if url is not None:
+            payload['url'] = url
+
+        if reporter is not None:
+            payload['reporter'] = reporter
+
+        if category is not None:
+            payload['category'] = category
+
+        if duration is not None:
+            payload['duration'] = duration
+
+        if self.verbose:
+            log.info(
+                'Sending Gerrit status for {change_id}/{revision_id}: data={data}',
+                change_id=change_id, revision_id=revision_id, data=payload)
+
+        return self._http.post('/'.join([
+            'changes', str(change_id), 'revisions', str(revision_id), 'verify-status~verifications']),
+            json=payload)
+
+    def formatDuration(self, duration):
```

Comment:
is it a part of the API?  (i think it's OK to override "private" methods as well)